### PR TITLE
feat(web): enable UTF-8 for names (AB, Members, Spaces)

### PR DESCRIPTION
**Pull request by vmosharova**
## What it solves

Resolves: [WA-2668](https://linear.app/safe-global/issue/WA-2707/adjust-frontend-to-cgw-changes),
Resolves: [WA-2646](https://linear.app/safe-global/issue/WA-2646/error-message-indicates-that-member-names-may-contain-underscores-but) as secondary
Resolves: [WA-2715](https://linear.app/safe-global/issue/WA-2715/address-book-and-accounts-ui-feedback) (related minor UI changes)
Related PR: [CGW 3178](https://github.com/safe-global/safe-client-gateway/pull/3178)

Resolves inconsistent and incomplete name validation across the web app. Names entered for address book contacts, Safe owners, workspaces, and workspace members were not validated consistently with the backend (safe-client-gateway), so users could submit names with disallowed/invisible characters, mismatched length rules, or get them rejected only at the server with raw error strings. CSV export was also not guarded against formula injection.

## How this PR fixes it

- Adds a shared, framework-agnostic name validation module in `@safe-global/utils` (`validation/names.ts`) that mirrors the CGW `name.schema.ts` — allowed character set, length bounds (default 3–30, address book 50, member 255), Unicode-aware code-point counting, NFC normalization, invisible-character stripping, and smart-punctuation folding. A ported test suite acts as the drift guard against the backend schema.
- Adds `escapeCsvFormula` in `@safe-global/utils` (`utils/csv.ts`), mirroring the CGW formula-escape util, to neutralize CSV/formula injection (`= + - @`, leading whitespace, full-width variants) on export.
- Wires the shared validation into `NameInput` (sanitize on blur, validate via RHF, focus-preserving tooltip for the "Invalid characters" case), the address book Entry/Export/Import dialogs, the Edit Owner dialog, and the workspace create/settings/member/invite/address-book forms.
- **CSV import for local address book now allows names with any non-empty value** — the strict character validation that previously blocked valid names with special characters has been relaxed; only empty names are rejected. Names with special characters (e.g. from external CSVs) are imported as-is.
- CSV import (workspace address book) still validates each row's name with the shared rules and reports the offending row and reason.
- Adds `getRtkQueryErrorMessage` / `applyBackendNameError` / `getBackendNameError` helpers so backend 422 validation errors surface inline on the right field, and transport-level errors (network/timeout/rate-limit/non-JSON body) show friendly copy instead of raw JS error strings.
- The "Add to workspace" button on local address book contacts now shows a human-readable error message instead of a raw error string when the server rejects the request, and is **disabled with an explanatory tooltip when the contact's name contains invalid characters** — preventing a server rejection before the user even tries.
- Fixes workspace identity section missing `maxLength` validation, preventing users from entering overly long workspace names without feedback.
- Fixes UI details: input keeps focus on invalid character entry, and the import preview hides entry count and connector line when the CSV is invalid.

### Additional small fixes
- E2E tests (smoke + regression) that were failing due to AB + Spaces changes

## How to test it

1. **Address book**: Add/edit a contact. Try names with emoji-only/invisible chars, disallowed symbols (`= < > {}`), and very long names — invalid input shows an inline "Invalid characters" label with a tooltip; valid input is sanitized (smart quotes → `'`, dashes normalized, trimmed) on blur.
2. **CSV import (local address book)**: Import a CSV containing names with special characters (e.g. `O'Brien`, `Müller`, names with commas) — these should import successfully. Only rows with completely empty names should be rejected.
3. **CSV import (workspace address book)**: Import a CSV with an invalid name on a row — the error names the row and reason; the entry count and connector line are hidden for an invalid file.
4. **CSV export**: Export an address book containing a name starting with `=`, `+`, `-`, `@`, or whitespace — the exported cell is prefixed with `'` so spreadsheets don't evaluate it as a formula.
5. **Owners / Workspaces**: Repeat name entry on the Edit Owner dialog, workspace creation, workspace settings, member info, invite, and workspace address book forms — all apply the same rules. Workspace identity name field should enforce `maxLength`.
6. **Backend errors**: Trigger a server-side 422 (e.g. a name the backend rejects) — the message appears inline on the field, not as a toast with a raw string. The "Add to workspace" button should also show a readable error message on failure.
7. **Add to workspace button**: Open a local address book contact whose name contains invalid characters — the "Add to workspace" button should be disabled and show a tooltip explaining why.

## Affected flows

- Address book: create / edit contact, CSV import (local), CSV export
- Safe settings: edit owner name
- Workspaces: create workspace onboarding, workspace settings (identity), add member, invite member (accept invite), workspace address book add/edit contact, add local contact to workspace

## Blast radius

- **Shared package `@safe-global/utils`**: new `validation/names.ts` (exported via `validation/index.ts`) and `utils/csv.ts`. New code, no changes to existing exports — but lives in a package consumed by both web and mobile.
- **`NameInput` (`apps/web/src/components/common/NameInput`)**: validation behavior, sanitize-on-blur, and label/tooltip rendering changed. Consumers: address book Entry dialog, Edit Owner dialog, workspace member/settings/identity/invite/address-book forms.
- **`FileUpload` (`apps/web/src/components/common/FileUpload`)**: invalid-state rendering (entry count / connector line) changed; used by the address book Import dialog.
- **`apps/web/src/utils/rtkQuery.ts`**: new error-message helpers; adopted by workspace mutation forms (create, update, add member, invite) and the "Add to workspace" button.
- **`ImportDialog/validation.ts`**: local AB CSV import name validation relaxed — only empty names now fail. Previously any name failing the strict character set was rejected.
- **`SpaceSettings/IdentitySection`**: `maxLength` now enforced on workspace name input.
- **`AddToWorkspaceButton`**: disabled state + tooltip added when contact name contains invalid characters; unit tests added.
- **CGW contract coupling**: `names.ts` and `csv.ts` are deliberate mirrors of backend schemas; their test suites guard against drift. If the backend schema changes, these must be updated in lockstep.

## Risks / not checked

- Did not exhaustively verify every Unicode edge case beyond the ported CGW test suite (e.g. exotic combining-mark sequences).
- Did not manually exercise every transport-error branch (timeout / rate-limit / non-JSON body) end-to-end against a live backend.
- Drift between these mirrors and the actual CGW schema is only caught if the backend schema itself changes and someone re-checks; there is no automated cross-repo check.
- Did not test the relaxed local CSV import against a broad range of real-world CSV exports from other wallet tools.

## Visual summary

https://github.com/user-attachments/assets/aa09711f-26b4-4933-ad03-2dfd9a7b90b9

```mermaid
flowchart TB
  subgraph Shared["@safe-global/utils (new)"]
    N["validation/names.ts<br/>sanitizeName · validateName"]
    C["utils/csv.ts<br/>escapeCsvFormula"]
  end

  subgraph Web["apps/web"]
    NI["NameInput"] --> N
    IMP_WS["Workspace AB CSV import"] --> N
    IMP_LOCAL["Local AB CSV import<br/>(relaxed: non-empty only)"]
    EXP["Address book CSV export"] --> C
    OWN["Edit Owner dialog"] --> NI
    AB["Address book Entry dialog"] --> NI
    WS["Workspace forms<br/>(create · settings · member · invite · address book)"] --> NI
    WS --> RTK["rtkQuery.ts<br/>applyBackendNameError · getRtkQueryErrorMessage"]
    ADD_BTN["AddToWorkspaceButton<br/>(disabled + tooltip on invalid name)"] --> RTK
    ADD_BTN --> N
  end

  CGW["safe-client-gateway schemas<br/>(name.schema · escape-csv-formula)"] -. mirrored & drift-guarded by tests .-> Shared
```

## Checklist

- [ ] I've tested the branch on mobile 📱
- [ ] I've documented how it affects the analytics (if at all) 📊
- [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻
- [ ] I've listed affected flows and blast radius, and named what I did not verify 🎯

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

### CODE REVIEW

**Comment by vmosharova** [comment]
@claude review

File: packages/utils/src/validation/names.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    result = result.replace(pattern, replacement)
+  }
+  return result.trim()
+}
+
+// Assumes value is already sanitized. Check order mirrors CGW.
+export const validateName = (value: string, opts?: { minLength?: number; maxLength?: number }): string | undefined => {
+  const min = opts?.minLength ?? NAME_MIN_LENGTH
+  const max = opts?.maxLength ?? NAME_MAX_LENGTH
+  const length = [...value].length // code points
```
Comment (TimDaub) [inline packages/utils/src/validation/names.ts:55]: What does `// code points` mean?

File: packages/utils/src/utils/csv.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Mirror of safe-client-gateway/src/modules/csv-export/csv-utils/escape-csv-formula.ts.
+// Triggers: ASCII = + - @, leading whitespace, and full-width ＝＋－＠ (folded by spreadsheets).
+const FORMULA_TRIGGERS = ['=', '+', '-', '@', ' ', '\t', '\r', '\n', '＝', '＋', '－', '＠']
+
+export const escapeCsvFormula = (value: unknown): string => {
```
Comment (TimDaub) [inline packages/utils/src/utils/csv.ts:5]: Same comment that I wrote on CGW PR also applies here since this file is mirrored: https://github.com/safe-global/safe-client-gateway/pull/3178#issuecomment-4797008227

File: packages/utils/src/validation/names.ts (hunk 1 above)
Comment (vmosharova) [reply packages/utils/src/validation/names.ts:55]: A code point is a Unicode value unit, and some chars have the letter + a mark (like `ñ`), so it looks like 1 char but in fact its `.length` might be 2.

File: packages/utils/src/utils/csv.ts (hunk 2 above)
Comment (vmosharova) [reply packages/utils/src/utils/csv.ts:5]: I agree that it might make sense to do some refactoring and use the formula escape function from the library on the CGW side, but I'm not sure what are trade-offs in adding this library as a dependency here 🤔

**Comment by tongyaosafe** [comment]
One comment:
Yesterday @compojoom pushed the feature "when adding local address book contact to space address book, the contact is **NOT** deleted from local address book".
Please make sure this feature is still there after solving merge conflicts.

**Comment by vmosharova** [comment]
Closing this PR as it was decided to split it into several smaller ones (https://github.com/safe-global/safe-wallet-monorepo/pull/8228, https://github.com/safe-global/safe-wallet-monorepo/pull/8227, more to follow)
